File: algorithm/88.merge-sorted-array.py
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: void Do not return anything, modify nums1 in-place instead.
        """
        
        left = m - 1
        right = n - 1
        total = m + n - 1

        while (left >= 0 and right >= 0):
            if nums1[left] < nums2[right]:
                nums1[total] = nums2[right]
                right -= 1
            else:
                nums1[total] = nums1[left]
                left -= 1
            total -= 1
        
        while (right >= 0):
            nums1[total] = nums2[right]
            right -= 1
            total -= 1

File: leetcode/88.merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int left = m - 1;
        int right = n - 1;
        int cur = m + n - 1;

        while(left >= 0 && right >=0)
        {
            if (nums1[left] >= nums2[right])
            {
                nums1[cur--] = nums1[left--];
            }
            else
            {
                nums1[cur--] = nums2[right--];
            }
        }   

        while(left >= 0)
        {
            nums1[cur--] = nums1[left--]; 
        }

        while(right >= 0)
        {
            nums1[cur--] = nums2[right--];
        }
    }
};
